src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -127,6 +144,26 @@
      */
     private final static String LDAP_READ_TIMEOUT = "com.sun.jndi.ldap.read.timeout";
 
+    /**
+     * The core pool size for the {@link ExecutorService}
+     */
+    private static final int corePoolSize = 4;
```

Comment:
should these be tunable via system properties? or perhaps should these be configurable in an advanced box? I suspect the answers are Yes, No... but I'd advise seeking others opinions